PulseSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java

```diff
@@ -0,0 +1,199 @@
+package com.ooyala.sample.players;
+
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.util.Log;
+import android.view.KeyEvent;
+
+import com.facebook.react.modules.core.DefaultHardwareBackBtnHandler;
+import com.ooyala.android.OoyalaNotification;
+import com.ooyala.android.OoyalaPlayer;
+import com.ooyala.android.PlayerDomain;
+import com.ooyala.android.skin.OoyalaSkinLayoutController;
+import com.ooyala.android.util.SDCardLogcatOoyalaEventsLogger;
+import com.ooyala.sample.utils.VideoItem;
+
+import java.util.Observable;
+import java.util.Observer;
+
+import static android.Manifest.permission.WRITE_EXTERNAL_STORAGE;
+import static android.content.pm.PackageManager.PERMISSION_GRANTED;
+
+/**
+ * This class asks permission for WRITE_EXTERNAL_STORAGE. We need it for automation hooks
+ * as we need to write into the SD card and automation will parse this file.
+ */
+public abstract class AbstractHookActivity extends Activity implements Observer, DefaultHardwareBackBtnHandler {
+  private static final int PERMISSIONS_REQUEST_WRITE_EXTERNAL_STORAGE = 1;
+  String TAG = this.getClass().toString();
+  final int CLICK_THROUGH_REQUEST = 11234;
+  protected OoyalaSkinLayoutController playerSkinLayoutController;
+  private SDCardLogcatOoyalaEventsLogger log = new SDCardLogcatOoyalaEventsLogger();
+
+  final String PCODE = "tlM2k6i2-WrXX1DE_b8zfhui_eQN";
```

Comment:
Move this `PCODE` and `DOMAIN` to the PulsePlayerActivity, you don't use them here.